### ISSUE TITLE
kr sends messages to wrong terminal screens (I think)

### ISSUE THREAD

**Issue by drcongo** [kryptco/kr#129]
I have iTerm2 on macOS, split into four panes, the top two are my Vagrant box, bottom right is the remote server, and bottom left local. Running a `git clone` in one of the Vagrant windows somehow triggers kr installed on my Mac, and the `Kryptonite ▶ Requesting SSH authentication from phone` message appears in the bottom right pane, which is my remote server.

None of this sounds to me like it should even be possible. But I'll happily make a screencast of it if need be.

**Comment by kcking** [comment]
Are any of the remote hosts you are connecting to marked as "unknown host" in Kryptonite?

Also how are you connected to the Vagrant box, through `ssh`?

**Comment by drcongo** [comment]
Yeah, I think they come up as "unknown host" on my watch, and Vagrant I'm connected via "vagrant ssh" which uses the default insecure private key. I should probably have also mentioned, that I don't have kr installed on the Vagrant box or the remote machines.

**Comment by kcking** [comment]
I see. When an "unknown host" is connected to, `krd` cannot determine which `ssh` process requested the signature. Currently, our workaround to this is to send the message to every `ssh` process (however each process never repeats a message). This still should not be happening from an `ssh` invocation within a vagrant machine (unless vagrant is doing something special).

Could you please post or email to us the output of `ssh -vv git@github.com` from your vagrant machine?

**Comment by kcking** [comment]
closing for now -- feel free to reopen with the verbose output!

**Comment by drcongo** [comment]
Sorry @kcking, I got hella busy. I did however do a bunch of testing and it's definitely a Vagrant thing. It seems it defers keys up to the host, so a vagrant machine is able to request my kr key from the host machine. Sounds like it's not really your issue though, apart from the unexpected (for the user) behaviour of the kr messages appearing in another connection screen so happy to leave this closed.
